src/hotspot/share/runtime/frame.cpp

```diff
@@ -1558,6 +1558,39 @@ void frame::describe(FrameValues& values, int frame_no, const RegisterMap* reg_m
 
 #endif
 
+/**
+ * Gets the caller frame of `fr` for thread `t`.
+ *
+ * @returns an invalid frame (i.e. fr.pc() === 0) if the caller cannot be obtained
+ */
+frame frame::next_frame(frame fr, Thread* t) {
```

Comment:
@fbredber  thanks for looking at this. I'm not trying to change anything here, all I did was move the code from VMError into frame. Terminology gets complex with frames and we already have the notion of "sender" rather than caller in this code.